lib/Repeat_Ride.dart

```diff
@@ -26,17 +71,64 @@ class _RepeatRideState extends State<RepeatRide> {
     fontSize: 18,
   );
 
-  final descriptionStyle = TextStyle(
-    color: Colors.grey,
-    fontWeight: FontWeight.w100,
-    fontSize: 13
-  );
+  final descriptionStyle =
+      TextStyle(color: Colors.grey, fontWeight: FontWeight.w100, fontSize: 13);
+
+  Widget _buildPickupTimeField() {
+    return TextFormField(
+      controller: pickUpCtrl,
+      decoration: InputDecoration(
+          labelText: 'Pickup Time',
+          hintText: 'Pickup Time',
+          labelStyle: TextStyle(color: Colors.black)),
+      validator: (input) {
+        if (input.isEmpty) {
+          return 'Please enter your pickup time';
+        }
+        return null;
+      },
+      onTap: () => selectPickUpTime(context),
+      style: TextStyle(color: Colors.black, fontSize: 15),
+    );
+  }
+
+  Widget _buildDropOffTimeField() {
+    return TextFormField(
+      controller: dropOffCtrl,
+      decoration: InputDecoration(
+          labelText: 'Drop-off Time',
+          hintText: 'Drop-off Time',
+          labelStyle: TextStyle(color: Colors.black)),
+      validator: (input) {
+        if (input.isEmpty) {
+          return 'Please enter your drop-off time';
+        }
+        return null;
+      },
+      onTap: () => selectDropOffTime(context),
+      style: TextStyle(color: Colors.black, fontSize: 15),
+    );
+  }
 
   final List<bool> isSelected = [false, false, false, false, false];
+  final List<String> days = ["M", "T", "W", "Th", "F"];
+
+  String setSelectedDays(String sDays) {
```

Comment:
Since you already keep the days selected in `isSelected`, you could just build the string once when the button is pressed rather than needing to iterate through all of the days and change the string every time a button is pressed--just a minor possible efficiency improvement